torch/_functorch/aot_autograd.py

```diff
@@ -3506,6 +3577,7 @@ def metadata_fn(*primals):
         metadata_fn,
         keep_input_mutations=meta.keep_input_mutations,
         is_train=meta.is_train,
+        is_subclass=True,
```

Comment:
A comment here explaining the situation would be great. The entire reason we're plumbing this is_subclass argument everywhere here is being it's painful to support input mutations on subclasses, and that info isn't easily available.